plugins/org.komodo.core/src/org/komodo/repository/RepositoryImpl.java

```diff
@@ -356,6 +348,18 @@ public RepositoryImpl( final Type type,
         this.id = id;
     }
 
+    private UnitOfWork verifyTransaction(final UnitOfWork uow, String name, boolean rollback) throws KException {
```

Comment:
This is a great method. Gets rid of lots of code. I'm wondering if we should also move into here the debug statement that usually follows a call to this method.